packages/flutter_tools/lib/src/device.dart

```diff
@@ -277,30 +276,57 @@ abstract class PollingDeviceDiscovery extends DeviceDiscovery {
   static const Duration _pollingTimeout = Duration(seconds: 30);
 
   final String name;
-  ItemListNotifier<Device> _items;
+  final Map<String, Device> _activeDevices = <String, Device>{};
+  final StreamController<Device> _addedController = StreamController<Device>.broadcast();
+  final StreamController<Device> _removedController = StreamController<Device>.broadcast();
+
   Timer _timer;
 
   Future<List<Device>> pollingGetDevices({ Duration timeout });
 
   void startPolling() {
-    if (_timer == null) {
-      _items ??= ItemListNotifier<Device>();
-      _timer = _initTimer();
+    _timer ??= _initTimer();
+  }
+
+  @visibleForTesting
+  Future<void> poll() async {
```

Comment:
Hoisted out so that we don't need a Timer. I think the general test solution to this sort of thing is some sort of injected Timer service that is external to each class doing the polling (see also, loggers/status spinners). Putting on my big-enterprise hat:

```
abstract class Pollable {
  FutureOr<void> poll();
}

abstract class PollService {
  PollSubscription register(Pollable pollable, Duration duration);
}
```

Could just be the isolation taking its toll though XD